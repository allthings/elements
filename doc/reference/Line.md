<!-- 
This is an auto-generated markdown. 
You can change it in "/Users/daniel/Dev/allthings/elements/src/Line/Line.tsx" and run build:docs to update this file.
-->
# Line
Just walking the line.

```example
<Line color="primary" />
```
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|color **(required)**||Color of the Line
