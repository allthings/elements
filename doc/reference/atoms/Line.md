<!-- 
This is an auto-generated markdown. 
You can change it in "src/atoms/Line.tsx" and run build:docs to update this file.
-->
# Line
Just walking the line.

```example
<Line color="primary" />
```
## Usage
| Name        | Type           | Description  |
| ----------- |:--------------:| ------------:|
|color **(required)**||Color of the Line
