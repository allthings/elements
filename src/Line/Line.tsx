import React from 'react'
import View from '../View'
import { css } from 'glamor'
import { colorCode } from '../utils/propTypes/color'

const line = (color: string) =>
  css({
    backgroundColor: color,
    height: '1px',
    width: '100%',
  })

/**
 * Just walking the line.
 *
 * ```example
 * <Line color="primary" />
 * ```
 */
const Line = ({
  color,
}: {
  /** Color of the Line */
  color: string
}) => (
  <View
    {...line(colorCode(color))}
    alignV="center"
    alignH="center"
    direction="column"
  />
)

export default Line
