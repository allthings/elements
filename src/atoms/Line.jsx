import React from 'react'
import View from '@allthings/react-view'
import { css } from 'glamor'
import { color, colorCode } from '../propTypes/color'

const line = color =>
  css({
    backgroundColor: color,
    height: '1px',
    width: '100%',
  })

/** Hello, I'm a Line */
const Line = ({ color }) => (
  <View
    {...line(colorCode(color))}
    alignV="center"
    alignH="center"
    direction="column"
  />
)

Line.propTypes = {
  /** Color of the Circle */
  color: color,
}

export default Line
