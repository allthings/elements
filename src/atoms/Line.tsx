import React from 'react'
import View from '../atoms/View'
import { css } from 'glamor'
import { colorCode } from '../propTypes/color'

const line = (color: string) =>
  css({
    backgroundColor: color,
    height: '1px',
    width: '100%',
  })

/**
 * Everything in elements is view! It's the component to align and layout things.
 *
 * ```example
 * <ThemeProvider>
 *   <View fill direction="row" alignH="end">
 *     <Text>Say Hello!</Text>
 *   </View>
 * </ThemeProvider>
 * ```
 */
const Line = ({
  color,
}: {
  /** Color of the Line */
  color: string
}) => (
  <View
    {...line(colorCode(color))}
    alignV="center"
    alignH="center"
    direction="column"
  />
)

export default Line
