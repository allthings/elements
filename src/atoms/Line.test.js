import React from 'react'
import renderer from 'react-test-renderer'
import Line from './Line'

test('Line renders correctly', () => {
  const tree = renderer.create(
    <Line />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
