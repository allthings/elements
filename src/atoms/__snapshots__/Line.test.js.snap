// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line renders correctly 1`] = `
.css-bawo03,
[data-css-bawo03] {
  height: 1px;
  width: 100%;
}

.css-yirehu,
[data-css-yirehu] {
  box-sizing: border-box;
  align-content: center;
  align-items: center;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  justify-content: center;
  flex: 0 0 auto;
  -webkit-align-content: center;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -webkit-flex: 0 0 auto;
}

<div
  data-css-bawo03=""
  data-css-yirehu=""
/>
`;
